SPEC.md

```diff
@@ -182,6 +182,37 @@ Contents give more information about the issue's check, including a description
 
 Some engines require the ability to refer to other source locations. For this reason, the Issue type has an optional `other_locations` field, which is an array of other `Location` items that this issue needs to refer to.
 
+## Versioning
+
+This specification is versioned. The current version is [in the repository](https://github.com/codeclimate/spec/blob/master/VERSION). Engines declare the version of the specification they are compatible with in the manifest file, described below.
+
+## Engine Config File
+
+All engines must include a `engine.json` file at `/engine.json`. This file includes information that is necessary for the analysis runtime and metadata about the engine. Here is an example config:
+
+```
+{
+  "name": "govet",
+  "description": "govet was created by the Go team at Google, and examines Go source code and reports suspicious constructs, and potential bugs.",
+  "maintainer": {
+    "name": "Michael R. Bernstein",
+    "email": "mrb@codeclimate.com",
+  },
+  "version": 3,
+  "spec_version": "0.0.1",
+}
+```
+
+The fields in the manifest include:
+
+* `name` (`String`) - the name of the package
+* `description` (`String`) - a description of the engine
+* `maintainer` (`Object`) - data about the engine maintainer
+  * `name` (`String`) - the name of the maintainer
+  * `email` (`String`) - the email address of the maintainer
+* `version` (`Integer`) - engine version
```

Comment:
Ah, sorry. In thinking about it, I think so long as we roll the cache on any change to the version, a string is fine, and engines can pick their own format.



—Bryan Helmkamp, Founder, Code Climate

bryan@brynary.com / 646-379-1810 / @brynary




Typed with thumbs.

On Fri, Sep 18, 2015 at 8:56 PM, Will Fleming <notifications@github.com>
wrote:

>> +    "name": "Michael R. Bernstein",
>> +    "email": "mrb@codeclimate.com",
>> +  },
>> +  "version": 3,
>> +  "spec_version": "0.0.1",
>> +}
>> +```
>> +
>> +The fields in the manifest include:
>> +
>> +* `name` (`String`) - the name of the package
>> +* `description` (`String`) - a description of the engine
>> +* `maintainer` (`Object`) - data about the engine maintainer
>> +  * `name` (`String`) - the name of the maintainer
>> +  * `email` (`String`) - the email address of the maintainer
>> +* `version` (`Integer`) - engine version
> Bryan had a strong preference for integer versioning. I'm not too fussed. For cache invalidation (which is what I care about), either works. 
>> On Sep 18, 2015, at 17:18, pat brisbin <notifications@github.com> wrote:
>> 
>> In SPEC.md:
>> 
>> > +    "name": "Michael R. Bernstein",
>> > +    "email": "mrb@codeclimate.com",
>> > +  },
>> > +  "version": 3,
>> > +  "spec_version": "0.0.1",
>> > +}
>> > +```
>> > +
>> > +The fields in the manifest include:
>> > +
>> > +* `name` (`String`) - the name of the package
>> > +* `description` (`String`) - a description of the engine
>> > +* `maintainer` (`Object`) - data about the engine maintainer
>> > +  * `name` (`String`) - the name of the maintainer
>> > +  * `email` (`String`) - the email address of the maintainer
>> > +* `version` (`Integer`) - engine version
>> Nit: by why are you forcing engines into an integer versioning scheme?
>> 
>> —
>> Reply to this email directly or view it on GitHub.
>> 
> ---
> Reply to this email directly or view it on GitHub:
> https://github.com/codeclimate/spec/pull/14/files#r39911907